replica/table.cc

```diff
@@ -1738,7 +1738,11 @@ future<db::replay_position> table::discard_sstables(db_clock::time_point truncat
             cf.refresh_compound_sstable_set();
         }
     };
-    auto p = make_lw_shared<pruner>(*this, *_compaction_group);
+    db::replay_position rp;
+    co_await parallel_foreach_compaction_group([this, truncated_at, &rp] (compaction_group& cg) -> future<> {
+
+    // FIXME: indentation
+    auto p = make_lw_shared<pruner>(*this, cg);
     co_await _cache.invalidate(row_cache::external_updater([p, truncated_at] {
```

Comment:
The alternative is to invalidate the cache by token range (in fact it supports it already). Here is makes sense to iterate sequentually, since invalidation is CPU intensive work and doing it in parallel just competes with queries.